cwl/src/main/scala/cwl/TypeAliases.scala

```diff
@@ -24,18 +24,33 @@ trait TypeAliases {
       StepInputExpressionRequirement :+:
       CNil
 
+  // TODO WOM: Record Schema as well as Directories are not included because they're not supported yet, although they should eventually be.
+  // Removing them saves some compile time when building decoders for this type (in CwlInputParsing)
+  type MyriadInputValuePrimitives = 
+    String :+:
+    Int :+:
+    Long :+:
+    File :+:
+    Float :+:
+    Double :+:
+    Boolean :+:
+    CNil
+
+  type MyriadInputValue =
+    MyriadInputValuePrimitives :+:
+      Array[MyriadInputValuePrimitives] :+:
+      CNil
+
   type MyriadInputType =
     CwlType :+:
       InputRecordSchema :+:
       InputEnumSchema :+:
       InputArraySchema :+:
-      String :+:
```

Comment:
Oh geez -_-
I guess we'll worry about what the semantic of an arbitrary string as a type is later..
Thanks I'll put it back

Reply:
@danbills by the way do you know how we could support arbitrary levels of nested arrays ? This doesn't seem like it could handle it.
This is not a very high priority but maybe later

Reply:
Recursive type in Array should be it.  I.e. `Array[MyriadInputType]`